src/app/story/directives/status-control.component.ts

```diff
@@ -0,0 +1,357 @@
+import { Component, DoCheck, Input, Output, EventEmitter, ViewChild, ElementRef } from '@angular/core';
+import { interval } from 'rxjs';
+import { skipWhile, mergeMap } from 'rxjs/operators';
+import { StoryModel } from 'app/shared';
+import { ToastrService, ModalService } from 'ngx-prx-styleguide';
+import { Router } from '@angular/router';
+import { Angulartics2 } from 'angulartics2';
+import { StoryStatus } from './status.component';
+
+interface PublishBoxButton {
+  text: string,
+  action: Function,
+  disabled?: boolean,
+  color?: 'red' | 'orange' | 'green' | 'plain',
+  icon?: string
+}
+@Component({
+  selector: 'publish-status-control',
+  template: `
+    <hr>
+    <div class="actions" *ngIf="story">
+      <prx-button [model]="story" working=0 disabled=0 plain=1
+        [visible]="isChanged || nextStatus !== currentStatus" (click)="discard()">Discard</prx-button>
+
+      <prx-button *ngIf="buttons" #dropdownButton
+        [model]="story"
+        visible="true"
+        [disabled]="buttons.primary.disabled"
+        [working]="working"
+        [red]="buttons.primary.color === 'red'"
+        [orange]="buttons.primary.color === 'orange'"
+        [green]="buttons.primary.color === 'green'"
+        [dropdown]="buttons.secondary"
+        (click)="buttons.primary.action()">
+        {{ buttons.primary.text }}
+        <div *ngIf="buttons.secondary" class="dropdown-menu-items">
```

Comment:
Perhaps a question for @brandonhundt - but I find it odd that there are 2 dropdowns in this box.  The upper "status" one, which is only used until you schedule or publish.  And this lower one, which allows you to unpublish or delete.  Is that intentional to distinguish between the actions?

I guess I was expecting to see "Save and Publish Now" and "Save and Schedule" down in this dropdown (depending on if the date you selected is in the future or in the past):

![image](https://user-images.githubusercontent.com/1410587/59376473-6465df80-8d0d-11e9-82f5-e7ae36df4822.png)

Reply:
I believe there was an issue with doing more than one action on save.

Reply:
Well, the API doesn't really support it.  But it seems like @sandikbarr has already worked around that limitation?  (If I edit the title, change draft -> published, and click "Publish Now" ... it both saves my title change and then publishes the episode).

Reply:
I worked thru the "Save and Publish" (dug thru the models and found what I needed to wait on after Save) so you can "Save & Publish" with a single button click. It takes a bit if your audio still needs to process, which is why I added the spinner to the Found 1 Problem button that shows while it needs to process.